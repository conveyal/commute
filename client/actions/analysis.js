import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
