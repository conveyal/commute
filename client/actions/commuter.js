import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addCommuter = createAction('add commuter')
export const createCommuter = ({commuter, groupId, organizationId}) => {
  commuter.id = uuid.v4()
  return [
    addCommuter({commuter, groupId, organizationId}),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete commuter')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteCommuter = ({commuterId, groupId, organizationId}) => [
  deleteLocally({commuterId, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]

const updateLocally = createAction('update commuter')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateCommuter = ({commuter, groupId, organizationId}) => [
  updateLocally({commuter, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]
