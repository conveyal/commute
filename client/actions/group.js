import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addGroup = createAction('add group')
const appendCommuters = createAction('append commuters')
export const appendToGroup = (newCommuters, groupId, organizationId) => {
  return [
    appendCommuters({ commuters: newCommuters, groupId, organizationId }),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}
export const createGroup = (newGroup, organizationId) => {
  newGroup.id = uuid.v4()
  newGroup.allAddressesGeocoded = false
  return [
    addGroup({ group: newGroup, organizationId }),
    push(`/organizations/${organizationId}/groups/${newGroup.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete group')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/group/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteGroup = (id, organizationId) => [
  deleteLocally({ groupId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]
