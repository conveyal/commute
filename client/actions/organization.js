import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addLocally = createAction('add organization')
export const createOrganization = (newOrganization) => {
  newOrganization.id = uuid.v4()
  newOrganization.analyses = []
  newOrganization.analysesById = {}
  newOrganization.sites = []
  newOrganization.sitesById = {}
  newOrganization.groups = []
  newOrganization.groupsById = {}
  return [
    addLocally(newOrganization),
    push(`/organizations/${newOrganization.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete organization')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/organization/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteOrganization = (id) => [deleteLocally(id), push('/')]

export const updateOrganization = (organization) => [
  setLocally(organization),
  push(`/organizations/${organization.id}`)
]
/* const updateOnServer = (organization) =>
  serverAction({
    url: `/api/organization/${organization.id}`,
    options: {
      body: JSON.stringify(organization),
      method: 'put'
    }
  }) */  // TODO: update on server

export const setLocally = createAction('set organization')
