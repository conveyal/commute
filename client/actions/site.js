import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addSite = createAction('add site')
export const createSite = (newSite, organizationId) => {
  newSite.id = uuid.v4()
  return [
    addSite({ organizationId, site: newSite }),
    push(`/organizations/${organizationId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete site')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteSite = (id, organizationId) => [
  deleteLocally({ siteId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const updateLocally = createAction('update site')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateSite = (site, organizationId) => [
  updateLocally({ organizationId, site }),
  push(`/organizations/${organizationId}`)
]
