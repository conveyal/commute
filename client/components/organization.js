import React, {Component, PropTypes} from 'react'
import {Button, ButtonGroup, Col, Grid, Row} from 'react-bootstrap'
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table'
import {Link} from 'react-router'

import Icon from './icon'
import {messages} from '../utils/env'
import {arrayCountRenderer} from '../utils/table'
import {actUponConfirmation} from '../utils/ui'

export default class Organization extends Component {
  static propTypes = {
    // dispatch
    deleteAnalysis: PropTypes.func,
    deleteGroup: PropTypes.func,
    deleteOrganization: PropTypes.func,
    deleteSite: PropTypes.func,

    // props
    organization: PropTypes.object
  }

  _analysisGroupNameRenderer = (cell, row) => {
    const organizationId = this.props.organization.id
    const group = this.props.organization.groupsById[row.siteId]
    return <Link to={`/organizations/${organizationId}/groups/${group.id}`}>{group.name}</Link>
  }

  _analysisNameRenderer = (cell, row) => {
    const organizationId = this.props.organization.id
    return <Link to={`/organizations/${organizationId}/analysis/${row.id}`}>{row.name}</Link>
  }

  _analysisSiteNameRenderer = (cell, row) => {
    const organizationId = this.props.organization.id
    const site = this.props.organization.sitesById[row.siteId]
    return <Link to={`/organizations/${organizationId}/groups/${site.id}`}>{site.name}</Link>
  }

  _analysisToolsRenderer = (cell, row) => {
    const doDelete = () => {
      this.props.deleteAnalysis(row.id, this.props.organization.id)
    }
    const onClick = () => actUponConfirmation(messages.analysis.deleteConfirmation, doDelete)
    return <Button bsStyle='danger' onClick={onClick}>Delete</Button>
  }

  _handleDelete = () => {
    const doDelete = () => this.props.deleteOrganization(this.props.organization.id)
    actUponConfirmation(messages.organization.deleteConfirmation, doDelete)
  }

  _groupNameRenderer = (cell, row) => {
    const organizationId = this.props.organization.id
    return <Link to={`/organizations/${organizationId}/groups/${row.id}/`}>{row.name}</Link>
  }

  _groupToolsRenderer = (cell, row) => {
    const doDelete = () => this.props.deleteGroup(row.id, this.props.organization.id)
    const onClick = () => actUponConfirmation(messages.group.deleteConfirmation, doDelete)
    return <Button bsStyle='danger' onClick={onClick}>Delete</Button>
  }

  _siteNameRenderer = (cell, row) => {
    const organizationId = this.props.organization.id
    return <Link to={`/organizations/${organizationId}/sites/${row.id}/`}>{row.name}</Link>
  }

  _siteToolsRenderer = (cell, row) => {
    const organizationId = this.props.organization.id
    const doDelete = () => this.props.deleteSite(row.id, organizationId)
    const onClick = () => actUponConfirmation(messages.site.deleteConfirmation, doDelete)
    return <div>
      <Button bsStyle='warning'>
        <Link to={`/organizations/${organizationId}/sites/${row.id}/edit`}>Edit</Link>
      </Button>
      <Button bsStyle='danger' onClick={onClick}>Delete</Button>
    </div>
  }

  render () {
    const {analyses, id: _id, groups, name, sites} = this.props.organization

    return (
      <Grid>
        <Row>
          <Col xs={12}>
            <h2><Icon type='shield' />{name}</h2>
            <ButtonGroup>
              <Button bsStyle='warning'>
                <Link to={`/organizations/${_id}/edit`}>Edit</Link>
              </Button>
              <Button bsStyle='danger' onClick={this._handleDelete}>Delete</Button>
            </ButtonGroup>
            <p>Below are this organization's sites, commuter groups and analyses.</p>
            <h3>Sites
              <Button className='pull-right'>
                <Link to={`/organizations/${_id}/sites/create`}>Create a new site <Icon type='building' /></Link>
              </Button>
            </h3>
            <p>A site is a location of a building or new address that you want to use as the centerpoint of your commutes.</p>
            <BootstrapTable data={sites}>
              <TableHeaderColumn dataField='id' isKey hidden />
              <TableHeaderColumn dataFormat={this._siteNameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataField='address'>Address</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._siteToolsRenderer}>Tools</TableHeaderColumn>
            </BootstrapTable>
            <h3>Commuter Groups
              <Button className='pull-right'>
                <Link to={`/organizations/${_id}/groups/create`}>Create a new group <Icon type='users' /></Link>
              </Button>
            </h3>
            <p>A commuter group is a list of commuters that can commute to a particular site.</p>
            <BootstrapTable data={groups}>
              <TableHeaderColumn dataField='id' isKey hidden />
              <TableHeaderColumn dataFormat={this._groupNameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataField='commuters' dataFormat={arrayCountRenderer}>Commuters</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._groupToolsRenderer}>Tools</TableHeaderColumn>
            </BootstrapTable>
            <h3>Analyses
              <Button className='pull-right'>
                <Link to={`/organizations/${_id}/analysis/create`}>Create a new analysis <Icon type='bar-chart' /></Link>
              </Button>
            </h3>
            <p>An analysis calculates commuting statistics for a pairing of a commuter group and site.</p>
            <BootstrapTable data={analyses}>
              <TableHeaderColumn dataField='id' isKey hidden />
              <TableHeaderColumn dataFormat={this._analysisNameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._analysisSiteNameRenderer}>Site</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._analysisGroupNameRenderer}>Group</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._analysisToolsRenderer}>Tools</TableHeaderColumn>
            </BootstrapTable>
          </Col>
        </Row>
      </Grid>
    )
  }
}
