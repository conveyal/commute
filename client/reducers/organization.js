import update from 'react-addons-update'

import {settings} from '../utils/env'
import {arrayToObj, fixedRound} from '../utils/reducers'

const METRICS = Object.keys(settings.metrics)

export const reducers = {

  /**
   * Organization Stuff
   */

  'add organization' (state, action) {
    return addEntities(state, 'organizations', [action.payload])
  },
  'delete organization' (state, action) {
    return deleteEntity(state, 'organizations', action.payload.id)
  },
  'set organization' (state, action) {
    return setOrganization(state, action.payload)
  },
  'set organizations' (state, action) {
    const organizations = [...action.payload]
    return {
      organizations,
      organizationsById: arrayToObj(action.payload)
    }
  },

  /**
   * Analysis Stuff
   */

  'add analysis' (state, action) {
    const {organizationId, analysis} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const updatedOrganization = addEntities(affectedOrganization, 'analyses', [analysis])
    return setOrganization(state, updatedOrganization)
  },
  'delete analysis' (state, action) {
    const {organizationId, analysisId} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const updatedOrganization = deleteEntity(affectedOrganization, 'analyses', analysisId)
    return setOrganization(state, updatedOrganization)
  },
  'receive mock calculated trips' (state, action) {
    const {analysisId, organizationId, trips} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const affectedAnalysis = affectedOrganization.analysesById[analysisId]
    let updatedAnalysis = update(affectedAnalysis, {
      trips: { $set: trips }
    })

    // calculate array of values for each metric
    const vals = {}
    const modes = Object.keys(trips[0]).filter((mode) => (['commuterId']).indexOf(mode) === -1)
    modes.forEach((mode) => {
      vals[mode] = {
        cost: [],
        distance: [],
        time: []
      }
    })
    trips.forEach((trip) =>
      modes.forEach((mode) =>
        METRICS.forEach((metric) => {
          vals[mode][metric].push(trip[mode][metric])
        })
      )
    )

    // sort arrays
    modes.forEach((mode) =>
      METRICS.forEach((metric) => {
        vals[mode][metric].sort()
      })
    )

    updatedAnalysis = update(updatedAnalysis, {
      tripVals: { $set: vals }
    })

    // calculate analysis summary metrics
    // Potential Savings is an overall average of the difference in cost between
    // the most likely trip and the driving trip
    const len = trips.length
    let totalCostDifference = 0
    let totalTravelTime = 0
    let totalDistance = 0
    trips.forEach((trip) => {
      totalCostDifference += trip.car.cost - trip.mostLikely.cost
      totalTravelTime += trip.mostLikely.time
      totalDistance += trip.mostLikely.distance
    })

    updatedAnalysis = update(updatedAnalysis, {
      summary: {
        $set: {
          avgTravelTime: fixedRound(totalTravelTime / len),
          avgDistance: fixedRound(totalDistance / len),
          savingsPerTrip: fixedRound(totalCostDifference / len),
          savingsPerTripYear: fixedRound(totalCostDifference / len * 260),
          savingsTotalPerDay: fixedRound(totalCostDifference),
          savingsTotalPerYear: fixedRound(totalCostDifference * 260)
        }
      }
    })

    const updatedOrganization = setEntity(affectedOrganization, 'analyses', updatedAnalysis)
    return setOrganization(state, updatedOrganization)
  },

  /**
   * Commuter Stuff
   */

  'add commuter' (state, action) {
    const {commuter, groupId, organizationId} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const affectedGroup = affectedOrganization.groupsById[groupId]
    const updatedGroup = addEntities(affectedGroup, 'commuters', [commuter])
    const updatedOrganization = setGroup(affectedOrganization, updatedGroup)
    return setOrganization(state, updatedOrganization)
  },
  'delete commuter' (state, action) {
    const {commuterId, groupId, organizationId} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const affectedGroup = affectedOrganization.groupsById[groupId]
    const updatedGroup = deleteEntity(affectedGroup, 'commuters', commuterId)
    const updatedOrganization = setGroup(affectedOrganization, updatedGroup)
    return setOrganization(state, updatedOrganization)
  },
  'update commuter' (state, action) {
    const {commuter, groupId, organizationId} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const affectedGroup = affectedOrganization.groupsById[groupId]
    const updatedGroup = setEntity(affectedGroup, 'commuters', commuter)
    const updatedOrganization = setGroup(affectedOrganization, updatedGroup)
    return setOrganization(state, updatedOrganization)
  },

  /**
   * Group Stuff
   */

  'add group' (state, action) {
    const {organizationId, group} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    group.commutersById = arrayToObj(group.commuters)
    const updatedOrganization = addEntities(affectedOrganization, 'groups', [group])
    return setOrganization(state, updatedOrganization)
  },
  'append commuters' (state, action) {
    const {commuters, groupId, organizationId} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const affectedGroup = affectedOrganization.groupsById[groupId]
    const updatedGroup = addEntities(affectedGroup, 'commuters', commuters)
    const updatedOrganization = setGroup(affectedOrganization, updatedGroup)
    return setOrganization(state, updatedOrganization)
  },
  'delete group' (state, action) {
    const {organizationId, groupId} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const updatedOrganization = deleteEntity(affectedOrganization, 'groups', groupId)
    return setOrganization(state, updatedOrganization)
  },

  /**
   * Site Stuff
   */

  'add site' (state, action) {
    const {organizationId, site} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const updatedOrganization = addEntities(affectedOrganization, 'sites', [site])
    return setOrganization(state, updatedOrganization)
  },
  'delete site' (state, action) {
    const {organizationId, siteId} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const updatedOrganization = deleteEntity(affectedOrganization, 'sites', siteId)
    return setOrganization(state, updatedOrganization)
  },
  'update site' (state, action) {
    const {organizationId, site} = action.payload
    const affectedOrganization = state.organizationsById[organizationId]
    const updatedOrganization = setEntity(affectedOrganization, 'sites', site)
    return setOrganization(state, updatedOrganization)
  }
}

// Utilities

/**
 * Helper to delete an item from a entityById object
 * @param  {String} id  The ID of the entity to delete
 * @return {Function}   A function that deletes the entity from
 *  the object and then returns that object
 */
const deleteFromMap = (id) => (map) => {
  delete map[id]
  return map
}

export const initialState = {
  organizations: [],
  organizationsById: {}
}

/**
 * Add entities to a collection
 *
 * @param {Object} parent           the parent object containing the collections
 * @param {String} entitiesName     base name for entity collection keys
 * @param {Array} entities          the entities to add
 * @return                          The updated parent
 */
const addEntities = (parent, entitiesName, entities) => {
  const entitiesByIdName = `${entitiesName}ById`
  let updatedParent = update(parent, {
    [entitiesName]: { $push: entities }
  })
  return update(updatedParent, {
    [entitiesByIdName]: { $set: arrayToObj(updatedParent[entitiesName]) }
  })
}

/**
 * Delete an entity within a collection
 *
 * @param {Object} parent           the parent object containing the collections
 * @param {String} entitiesName     base name for entity collection keys
 * @param {Object} entityId         ID of the entity to delete
 * @return                          The updated parent
 */
const deleteEntity = (parent, entitiesName, entityId) => {
  const entitiesByIdName = `${entitiesName}ById`
  let updatedParent = update(parent, {
    [entitiesByIdName]: { $apply: deleteFromMap(entityId) }
  })
  return update(updatedParent, {
    [entitiesName]: { $set: Object.values(updatedParent[entitiesByIdName]) }
  })
}

/**
 * Helper to update an entity within a collection
 *
 * @param {Object} parent           the parent object containing the collections
 * @param {String} entitiesName     base name for entity collection keys* @param {Object} updatedEntity    entity to update
 * @return                          The updated parent
 */
const setEntity = (parent, entitiesName, updatedEntity) => {
  const entitiesByIdName = `${entitiesName}ById`
  let updatedParent = update(parent, {
    [entitiesByIdName]: {
      [updatedEntity.id]: {
        $set: updatedEntity
      }
    }
  })
  return update(updatedParent, {
    [entitiesName]: {
      $set: Object.values(updatedParent[entitiesByIdName])
    }
  })
}

const setGroup = (organization, group) => {
  return setEntity(organization, 'groups', group)
}

const setOrganization = (state, organization) => {
  return setEntity(state, 'organizations', organization)
}
