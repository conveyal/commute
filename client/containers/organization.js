import {connect} from 'react-redux'

import {deleteAnalysis} from '../actions/analysis'
import {deleteGroup} from '../actions/group'
import {deleteOrganization} from '../actions/organization'
import {deleteSite} from '../actions/site'
import Organization from '../components/organization'

function mapStateToProps (state, props) {
  const {organization} = state
  const {params} = props
  const currentOrganization = organization.organizationsById[params.organizationId]
  return {organization: currentOrganization}
}

function mapDispatchToProps (dispatch) {
  return {
    deleteAnalysis: (id, organizationId) => dispatch(deleteAnalysis(id, organizationId)),
    deleteGroup: (id, organizationId) => dispatch(deleteGroup(id, organizationId)),
    deleteOrganization: (id) => dispatch(deleteOrganization(id)),
    deleteSite: (id, organizationId) => dispatch(deleteSite(id, organizationId))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Organization)
