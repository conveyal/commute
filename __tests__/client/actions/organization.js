/* global describe, expect, it */

import {expectCreateAction, expectDeleteOrganization} from '../../test-utils/actions'
import {simpleOrganization} from '../../test-utils/mock-data'

import * as organization from '../../../client/actions/organization'

describe('actions > organization', () => {
  it('create organization should work', () => {
    const data = {
      name: 'mockOrg'
    }
    const result = organization.createOrganization(data)

    expectCreateAction(result)
  })

  it('update organization should work', () => {
    const result = organization.updateOrganization(simpleOrganization)

    expect(result.length).toBe(2)
    expect(result).toMatchSnapshot()
  })

  it('delete organization should work', () => {
    const result = organization.deleteOrganization('1')

    expectDeleteOrganization(result)
  })
})
