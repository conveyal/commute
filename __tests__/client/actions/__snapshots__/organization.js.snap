exports[`actions > organization create organization should work 1`] = `
Object {
  "payload": Object {
    "analyses": Array [],
    "analysesById": Object {},
    "groups": Array [],
    "groupsById": Object {},
    "name": "mockOrg",
    "sites": Array [],
    "sitesById": Object {},
  },
  "type": "add organization",
}
`;

exports[`actions > organization create organization should work 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organizations/new-entity-id",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`actions > organization delete organization should work 1`] = `
Array [
  Object {
    "payload": "1",
    "type": "delete organization",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`actions > organization update organization should work 1`] = `
Array [
  Object {
    "payload": Object {
      "analyses": Array [],
      "analysesById": Object {},
      "contact": "someone",
      "email": "abc@def.ghi",
      "groups": Array [],
      "groupsById": Object {},
      "id": "1",
      "logo_url": "https://placekitten.com/80/80",
      "main_url": "https://placekitten.com/",
      "name": "Mock Organization",
      "sites": Array [],
      "sitesById": Object {},
    },
    "type": "set organization",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/organizations/1",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;
